src/app/abstractexecutablepage.cpp

```diff
@@ -77,19 +75,30 @@ AbstractExecutablePage::~AbstractExecutablePage() {}
 
 void AbstractExecutablePage::runTasks()
 {
-    cleanUpUi();
+    if (isServiceActive())
+    {
+        cleanUpUi();
 
-    runner->resetStopFlag();
+        runner->resetStopFlag();
 
-    workingThread = new QThread();
+        workingThread = new QThread();
```

Comment:
I have missed this earlier better name for this variable will be workerThread not workingThread.